lib/metasploit/framework/login_scanner/ssh.rb

```diff
@@ -94,8 +94,12 @@ def attempt_login(credential)
 
           unless result_options.has_key? :status
             if ssh_socket
-              proof = gather_proof unless skip_gather_proof
-              result_options.merge!(status: Metasploit::Model::Login::Status::SUCCESSFUL, proof: proof)
+              begin
+                proof = gather_proof unless skip_gather_proof
+                result_options.merge!(status: Metasploit::Model::Login::Status::SUCCESSFUL, proof: proof)
+              rescue => e
+                result_options.merge!(status: Metasploit::Model::Login::Status::INCORRECT, proof: e)
+              end
```

Comment:
I think if we've reached this point, the authentication was successful. We don't want to log the status as incorrect, that could cause additional login attempts to be made. We just want to log the exception info and set `proof` to nil since that's what actually failed to be recovered.

```suggestion
              begin
                proof = gather_proof unless skip_gather_proof
              rescue StandardError => e
                elog('Failed to gather SSH proof', error: e)
                proof = nil
              end
              result_options.merge!(status: Metasploit::Model::Login::Status::SUCCESSFUL, proof: proof)
```